src/main/java/uk/gov/hmcts/reform/et/syaapi/service/AssignCaseToLocalOfficeService.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.hmcts.reform.et.syaapi.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.dwp.regex.InvalidPostcodeException;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.items.RespondentSumTypeItem;
+import uk.gov.hmcts.reform.et.syaapi.helper.EmployeeObjectMapper;
+import uk.gov.hmcts.reform.et.syaapi.models.CaseRequest;
+
+import java.util.List;
+
+import static uk.gov.hmcts.ecm.common.model.helper.TribunalOffice.getCaseTypeId;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.DEFAULT_TRIBUNAL_OFFICE;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AssignCaseToLocalOfficeService {
+    private final PostcodeToOfficeService postcodeToOfficeService;
+
+    /**
+     * Assigns case to local office. Returns @{@link CaseData} object.
+     *
+     * @param caseRequest which would be in json format
+     * @return @link CaseData
+     */
+    public CaseData convertCaseRequestToCaseDataWithTribunalOffice(CaseRequest caseRequest) {
+        CaseData caseData = EmployeeObjectMapper.mapRequestCaseDataToCaseData(caseRequest.getCaseData());
+        List<RespondentSumTypeItem> respondentSumTypeList = caseData.getRespondentCollection();
+        String managingOffice = null;
+        if (caseData.getClaimantWorkAddress() != null
+            && caseData.getClaimantWorkAddress().getClaimantWorkAddress() != null
+            && StringUtils.isNotBlank(caseData.getClaimantWorkAddress().getClaimantWorkAddress().getPostCode())) {
+            managingOffice = getManagingOffice(
+                caseData.getClaimantWorkAddress().getClaimantWorkAddress().getPostCode());
+        } else if (!CollectionUtils.isEmpty(respondentSumTypeList)) {
+            for (RespondentSumTypeItem respondentSumTypeItem : respondentSumTypeList) {
+                if (respondentSumTypeItem.getValue() != null
+                    && respondentSumTypeItem.getValue().getRespondentAddress() != null
+                    && StringUtils.isNotBlank(respondentSumTypeItem.getValue().getRespondentAddress().getPostCode())) {
+                    managingOffice = getManagingOffice(
+                        respondentSumTypeItem.getValue().getRespondentAddress().getPostCode());
+                    break;
+                }
+            }
+        }
+        caseData.setManagingOffice(managingOffice);
+        return caseData;
+    }
+
+    private String getManagingOffice(String postcode) {
+        try {
+            return postcodeToOfficeService.getTribunalOfficeFromPostcode(postcode)
+                .orElse(DEFAULT_TRIBUNAL_OFFICE).getOfficeName();
```

Comment:
I'd like to check with @hsjhita what we want to return when a postcode is not provided or managing office is not found. I don't think we should be returning the default office we currently have defined in the code (London South)

Reply:
You'll have to check with Dan - I know we used to have a special code for this and I believe it used to default to Scotland but best check with Dan. Cases should still be able to be submitted without a postcode so it might just be a case of finding out what the default office should be

Reply:
Changed to "Unassigned".